// Copyright 2011, Tencent Inc.
// Author: Yi Wang <yiwang@tencent.com>
//
// This .proto file defines a sample RPC service for testing and
// demonstration.  During compilation, Blade
// (common/builder/blade/blade) would invoke protoc to generate a mock
// service stub, in addition to other stuff, in mock_rpc.pb.h.  This
// mock can then be used to test client library of the service.
//
// For the demonstration of using the mock in unittest, please refer
// to mock_rpc_test.cc.

import "poppy/rpc_option.proto";

package poppy.examples.mock_rpc;

message WordCountRequest {
    required string text = 1;
}

message WordCountResponse {
    required int32 count = 1;
}

service WordCountService {
    option (poppy.service_timeout) = 1000;
    rpc Count(WordCountRequest) returns(WordCountResponse);
}
